salt/client/ssh/__init__.py

```diff
@@ -797,6 +799,8 @@ def run(self, jid=None):
                     data = {'return': data}
                 if 'id' not in data:
                     data['id'] = id_
+                if 'fun' not in data:
+                    data['fun'] = fun
```

Comment:
This can be just written as:

```python
data.setdefault("fun", fun)
```

But I am still not sure this is the right place of doing it. In `handle_ssh` the function might not appear due to various reasons there as well as the next element is just a text, so the data was just created above in line 682. In that case I would rather find out why `fun` key did not appeared at all and if it is a bug, then should be done in the mentioned method instead.

Maybe @thatch45 might help here.

Reply:
This looks right since this data is only part of the event it should not interfere with any other aspect of execution. Using  data.setdefault("fun", fun) would also be an acceptable approach but it is not something that I am too concerned with here.

Reply:
My main concern is why _sometimes_ "fun" and "Id" in this case needs to be checked. Shouldn't we rather move that above where "data" variable is redefined from string type to dict instead? I mean:

```python
if isinstance(...
    data = {"id": ...., "fun":...
```
Instead of double-checking.